front/lib/actions/mcp_metadata.ts

```diff
@@ -224,11 +229,19 @@ export function extractMetadataFromTools(tools: Tool[]): MCPToolType[] {
 export async function fetchRemoteServerMetaDataByURL(
   auth: Authenticator,
   url: string
-): Promise<Omit<MCPServerType, "id">> {
-  const mcpClient = await connectToMCPServer(auth, {
-    type: "remoteMCPServerUrl",
-    remoteMCPServerUrl: url,
-  });
+): Promise<Result<Omit<MCPServerType, "id">, Error>> {
+  let mcpClient: Client;
+
+  try {
+    mcpClient = await connectToMCPServer(auth, {
+      type: "remoteMCPServerUrl",
+      remoteMCPServerUrl: url,
+    });
+  } catch (e: unknown) {
```

Comment:
We are not supposed to try catch our own methods why isn't connectToMCPServrr not using a Result ooc?

Reply:
it is now :) better to do that indeed